custom_components/localtuya/coordinator.py

```diff
@@ -347,9 +347,11 @@ async def close(self):
 
         tasks = [self._task_shutdown_entities, self._task_reconnect, self._task_connect]
         pending_tasks = [task for task in tasks if task and task.cancel()]
-        pending_tasks += [self.abort_connect()]
+        pending_tasks += []
```

Comment:
This does affect parent devices, why not move sub_devices close block above tasks cancel?

Reply:
The primary goal of the previous PR was to shutdown the whole LocalTuya fully and efficiently.

The primary goal of this PR is to fix the problem: calling `abort_connect()` before closing the sub-devices causes
 full featured `disconnected()` of sub-devices, including launching their `_task_reconnect` tasks, which is a waste of CPU time.

Now back to your question:

Doing close of sub-device before stopping those 3 tasks allows the 3 tasks to continue running during `await`s for sub-devices `close()` calls. This way the 3 tasks do redundant jobs, which is inefficient.

What may slightly improve the preformance is to put the sub-devices close loop _between_ cancelling the 3 tasks and `asyncio.gather` call for them. But it's a subtle improvement. Also, the current order has more predictable result.